io/shared/src/test/scala/fs2/io/net/tcp/SocketSuite.scala

```diff
@@ -158,6 +159,16 @@ class SocketSuite extends Fs2Suite with SocketSuitePlatform {
         .drain
     }
 
+    test("errors - should be captured in the effect") {
+      (for {
+        bindAddress <- Network[IO].serverResource(Some(ip"127.0.0.1")).use(s => IO.pure(s._1))
+        _ <- Network[IO].client(bindAddress).use(_ => IO.unit)
+      } yield ()).attempt.map(r => assert(r.isLeft)) >> (for {
+        bindAddress <- Network[IO].serverResource(Some(ip"127.0.0.1")).map(_._1)
+        _ <- Network[IO].serverResource(Some(bindAddress.host), Some(bindAddress.port))
+      } yield ()).attempt.use(r => IO(assert(r.isLeft)))
```

Comment:
May want to document what errors this is checking for.

Reply:
Good idea, done.